packages/react/src/lib/accessibility/FocusZone/FocusTrapZone.tsx

```diff
@@ -58,22 +59,24 @@ export class FocusTrapZone extends React.Component<FocusTrapZoneProps, {}> {
 
   componentDidMount(): void {
     FocusTrapZone._focusStack.push(this)
-    const { disableFirstFocus = false } = this.props
-
-    this._previouslyFocusedElementOutsideTrapZone = this._getPreviouslyFocusedElementOutsideTrapZone()
+    this._bringFocusIntoZone()
+    this._hideContentFromAccessibilityTree()
+  }
 
-    if (
-      !this._root.current.contains(this._previouslyFocusedElementOutsideTrapZone) &&
-      !disableFirstFocus
-    ) {
-      this._findElementAndFocusAsync()
+  componentDidUpdate(): void {
+    if (!this.props.forceFocusInsideTrapOnComponentUpdate) {
+      return
     }
 
-    this._hideContentFromAccessibilityTree()
+    const activeElement = document.activeElement as HTMLElement
+    // if after componentDidUpdate focus is not inside the focus trap, bring it back
+    if (!this._root.current.contains(activeElement)) {
+      this._bringFocusIntoZone()
+    }
   }
 
   render(): JSX.Element {
-    const { className, forceFocusInsideTrap, ariaLabelledBy } = this.props
+    const { className, forceFocusInsideTrapOnOutsideFocus, ariaLabelledBy } = this.props
```

Comment:
should we use these new props in popupFocusTrapBehavior?

Reply:
if we want it by default then probably set this prop to "true" in FocusTrapZone. But I am not sure if want this by default, maybe better to allow a user to do it through customizing the behavior